applications/FSIapplication/python_scripts/partitioned_embedded_fsi_base_solver.py

```diff
@@ -0,0 +1,665 @@
+            err_msg = 'Level set type is: \'' + self.level_set_type + '\'. Expected \'continuous\' or \'discontinuous\'.'
+            raise Exception(err_msg)
+
+        # Set the volume model part name in which the distance modification is applied to
+        volume_model_part_name = self.settings["fluid_solver_settings"]["volume_model_part_name"].GetString()
+        distance_modification_settings["model_part_name"].SetString(volume_model_part_name)
+
+        # Construct and return the distance modification process
+        return KratosFluid.DistanceModificationProcess(self.model, distance_modification_settings)
+
+    def _get_embedded_skin_utility(self):
+        if not hasattr(self, '_embedded_skin_utility'):
+            self._embedded_skin_utility = self._create_embedded_skin_utility()
+        return self._embedded_skin_utility
+
+    def _create_embedded_skin_utility(self):
+        if self.domain_size == 2:
+            return KratosMultiphysics.EmbeddedSkinUtility2D(
+                self.GetFluidComputingModelPart(),
+                self._get_embedded_skin_utility_model_part(),
+                self.level_set_type)
+        elif self.domain_size == 3:
+            return KratosMultiphysics.EmbeddedSkinUtility3D(
+                self.GetFluidComputingModelPart(),
+                self._get_embedded_skin_utility_model_part(),
+                self.level_set_type)
+        else:
+            raise Exception("Domain size expected to be 2 or 3. Got " + str(self.domain_size))
+
+    def _get_embedded_skin_utility_model_part(self):
+        if not hasattr(self, '_embedded_skin_utility_model_part'):
+            self._embedded_skin_utility_model_part = self._create_embedded_skin_utility_model_part()
+        return self._embedded_skin_utility_model_part
+
+    def _create_embedded_skin_utility_model_part(self):
+        embedded_skin_utility_skin_model_part_name = "EmbeddedSkinUtilityModelPart"
+        embedded_skin_utility_skin_model_part =self.model.CreateModelPart(embedded_skin_utility_skin_model_part_name)
+        embedded_skin_utility_skin_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
+        if (self.level_set_type == "continuous"):
+            embedded_skin_utility_skin_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PRESSURE)
+        elif (self.level_set_type == "discontinuous"):
+            embedded_skin_utility_skin_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.POSITIVE_FACE_PRESSURE)
+            embedded_skin_utility_skin_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NEGATIVE_FACE_PRESSURE)
+        else:
+            err_msg = 'Level set type is: \'' + self.level_set_type + '\'. Expected \'continuous\' or \'discontinuous\'.'
+            raise Exception(err_msg)
+        return embedded_skin_utility_skin_model_part
+
+    def _get_structure_interface_model_part_name(self):
+        str_int_list = self.settings["coupling_settings"]["structure_interfaces_list"]
```

Comment:
why is it a list if it cannot be more than one?

Reply:
The GUI allows to select more than one model part as structure interface to consider double-sided surfaces in the body-fitted case. Since there is no embedded FSI GUI yet, I decided to keep the I/O as similar as possible to the body-fitted case.

In any case, this is the very first implementation of the solver. I'm not sure at all if we should allow the user to select more than one model part as interface and internally to merge them in a unique model part.